lib/SyTest/Homeserver.pm

```diff
@@ -546,16 +548,34 @@ sub _run_command
    my %params = @_;
 
    my $cmd = $params{command}[0];
+   my $output = $self->{output};
+
+   my $diag = sub {
+      $output->diag( "\e[1;35m[$cmd]\e[m: $_[0]" );
+   };
+
+   my $on_output = sub {
+      my ( $stream, $buffref, $eof) = @_;
+      while ( $$buffref =~ s/^(.*)\n// ) {
+         &$diag($1);
+      }
+      if( $eof && $$buffref ) {
+         &$diag($$buffref);
+      }
```

Comment:
Is this bit for printing out the final line without a `\n` at the end?

Reply:
yeah, exactly. I'll add a comment.